Source/WebCore/dom/FullscreenManager.cpp

```diff
@@ -588,23 +585,17 @@ void FullscreenManager::dispatchFullscreenChangeOrErrorEvent(Deque<GCReachableRe
     }
 }
 
-void FullscreenManager::adjustFullscreenElementOnNodeRemoval(Node& node, Document::NodeRemoval nodeRemoval)
+void FullscreenManager::exitRemovedFullscreenElementIfNeeded(Element& element)
 {
-    auto fullscreenElement = fullscreenOrPendingElement();
-    if (!fullscreenElement)
+    if (!element.hasFullscreenFlag())
         return;
 
-    bool elementInSubtree = false;
-    if (nodeRemoval == Document::NodeRemoval::ChildrenOfNode)
-        elementInSubtree = fullscreenElement->isDescendantOf(node);
-    else
-        elementInSubtree = (fullscreenElement == &node) || fullscreenElement->isDescendantOf(node);
-
-    if (elementInSubtree) {
-        INFO_LOG(LOGIDENTIFIER, "Ancestor of fullscreen element removed; exiting fullscreen");
-        fullscreenElement->setContainsFullScreenElementOnAncestorsCrossingFrameBoundaries(false);
-        cancelFullscreen();
-    }
+    auto fullscreenElement = fullscreenOrPendingElement();
+    if (fullscreenElement && fullscreenElement == &element) {
```

Comment:
No need for the null check here. A null pointer will never be `== &element`. So:
```
if (fullscreenOrPendingElement() == &element)
```